### ISSUE TITLE
Add closeHttpClient to MinioAsyncClient.Builder

### ISSUE THREAD

**Issue by SerhiiKryvokin** [minio/minio-java#1586]
In PR #1546 the logic was added to close `httpClient` when closing minio client. 
However, if a custom `httpClient` is passed to the builder then `closeHttpClient` is set to `false`

Here is the code from `MinioAsyncClient.java`
```
boolean closeHttpClient = false;
  if (this.httpClient == null) {
    this.httpClient =
            HttpUtils.newDefaultHttpClient(
                    DEFAULT_CONNECTION_TIMEOUT, DEFAULT_CONNECTION_TIMEOUT, DEFAULT_CONNECTION_TIMEOUT);
    closeHttpClient = true;
}
```

So we set `closeHttpClient` to `true` only when `httpClient` is not passed to the builder and we create a default one  

My proposal is to allow to pass `closeHttpClient` to the builder and even make it `true` by default
I can create PR for this if the idea is approved

**Comment by balamurugana** [comment]
Yes. For custom http client, we do not set to close because it is user's responsibility when to close it.  You could add a new builder method like `public Builder httpClient(OkHttpClient httpClient, boolean close)`.

**Comment by SerhiiKryvokin** [comment]
Solved in PR #1587
